invokeai/backend/model_management/model_probe.py

```diff
@@ -115,6 +112,9 @@ def get_model_type_from_checkpoint(cls, model_path: Path, checkpoint: dict)->Mod
             return ModelType.TextualInversion
```

Comment:
Suggestions:
1) Add `.pth` extension(saw in checkpoint controlnet files)
2) Check for diffusers textual inversion lastly as it lest accurate detection
3) Check in loop to not iterate through all definitions in each check
4) Use `from .models.base import read_checkpoint_meta` for faster and less memory reading as there no need in tensors data(weights). 
  Add to this function additional model check to make definition like - `def read_checkpoint_meta(model_path: Union[str, Path], scan: bool = False)`
5) Decide what we do on failed detection, as there inconsistency - `get_model_type_from_checkpoint` returns None on fail, but `get_model_type_from_folder` raise Exception on fail

```python
def read_checkpoint_meta(path: Union[str, Path], scan: bool = False):
    if path.endswith(".safetensors"):
        try:
            checkpoint = _fast_safetensors_reader(path)
        except:
            # TODO: create issue for support "meta"?
            checkpoint = safetensors.torch.load_file(path, device="cpu")
    else:
        if scan:
            scan_result = scan_file_path(checkpoint)
            if scan_result.infected_files != 0:
                raise Exception(f"The model file \"{path}\" is potentially infected by malware. Aborting import.")
        checkpoint = torch.load(path, map_location=torch.device("meta"))
    return checkpoint
```

```python
    @classmethod
    def get_model_type_from_checkpoint(cls, model_path: Path, checkpoint: dict) -> ModelType:
        if model_path.suffix not in ('.bin','.pt','.ckpt','.safetensors','.pth'):
            return None

        if model_path.name == "learned_embeds.bin":
            return ModelType.TextualInversion

        checkpoint = checkpoint or read_checkpoint_meta(model_path, scan=True)
        checkpoint = checkpoint.get("state_dict", checkpoint)

        for key, tensor in checkpoint:
            if any(key.startswith(v) for v in {"cond_stage_model.", "first_stage_model.", "model.diffusion_model."}):
                return ModelType.Pipeline
            elif any(key.startswith(v) for v in {"encoder.conv_in", "decoder.conv_in"}):
                return ModelType.Vae
            elif any(key.startswith(v) for v in {"lora_te_", "lora_unet_"}):
                return ModelType.LoRA
            elif any(key.startswith(v) for v in {"control_model", "input_blocks"}):
                return ModelType.ControlNet
            elif key in {"emb_params", "string_to_param"}:
                return ModelType.TextualInversion

        else:
            # diffusers-ti
            if len(checkpoint) < 10 and all(isinstance(v, torch.Tensor) for v in checkpoint.values()):
                return ModelType.TextualInversion
        
        # give up
        return None
```

Reply:
Thank you for the review. I've accepted the proposed improvements with a few modifications necessary to get them to work properly.